boa/src/builtins/regexp/mod.rs

```diff
@@ -68,32 +66,28 @@ impl RegExp {
     pub(crate) const LENGTH: usize = 2;
 
     /// Create a new `RegExp`
-    pub(crate) fn make_regexp(
-        this: &mut Value,
-        args: &[Value],
-        ctx: &mut Interpreter,
-    ) -> ResultValue {
+    pub(crate) fn make_regexp(this: &Value, args: &[Value], ctx: &mut Interpreter) -> ResultValue {
         if args.is_empty() {
             return Err(Value::undefined());
         }
```

Comment:
I think we should use this opportunity to get the value. Something like:
```rust
let arg = match args.get(0).ok_or_else(Value::undefined)?;
```

And then, down below, we could use:
```rust
match *arg {
```

This would avoid that indexing, and the clippy lint exception.